preciceconfigchecker/rule.py

```diff
@@ -1,27 +1,55 @@
-from typing import List, Callable
-from violation import Violation, Problem
+from abc import ABC, abstractmethod
+from typing import List
 from severity import Severity
 
 
-class Rule:
-    def __init__(self, problem:Problem, severity:Severity = Severity.INFO) -> None:
-        self.problem = problem
+class Rule(ABC):
+    numbers:int = 0 #static attribute: do not use 'self.numbers', use 'Rule.numbers'
```

Comment:
What does this do? Name should identify usage. 
Is this for “naming” rules? Like Rule 1 = Data not found etc ?

Reply:
It's for output. Every violation counts, regardless of the rules. Is part of the formatting.

I can also just rename the variable to 'number_of_violations'.

Reply:
That would be nice

Reply:
done